aspnetcore/grpc/index.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Introduction to gRPC on ASP.NET Core
+author: juntaoluo
+description: Learn how to build gRPC services with Kestrel server and the ASP.NET Core stack.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: johluo
+ms.date: 02/26/2019
+uid: grpc/index
+---
+# Introduction to gRPC on ASP.NET Core
+
+By [John Luo](https://github.com/juntaoluo)
+
+gRPC is a language agnostic, high-performance Remote Procedure Call (RPC) framework incubated by the Cloud Native Computing Foundation (CNCF). For more on gRPC fundamentals, see the [gRPC documentation page](https://grpc.io/docs/). While a C# implementation is currently available on the official [gRPC page](https://grpc.io/docs/quickstart/csharp.html), the current implementation relies on the native library written in C (grpc C Core).
+
+Work is currently in progress to provide a new implementation based on the Kestrel HTTP server and the ASP.NET Core stack that is fully managed. This document provides an introduction to building gRPC services with this new implementation.
+
+[!INCLUDE[](~/includes/net-core-prereqs-all-3.0.md)]
+
+## Create a gRPC project
+
+# [Visual Studio](#tab/visual-studio)
+
+See [Get started with gRPC Services](xref:tutorials/grpc/grpc-start) for detailed instructions on how to create a gRPC project.
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+Run `dotnet new grpc` from the command line.
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+Run `dotnet new grpc` from the command line.
+
+Open the generated *.sln* file from Visual Studio for Mac.
+
+---
+
+## `.proto` file
+
+gRPC uses a design-first approach to API development. Protocol buffers (protobuf) are used as the Interface Design Language (IDL) by default. As a result, `.proto` file serve as the definition of the gRPC service and messages sent between clients and servers. For more information on the syntax of protobuf files, visit the [official documentation (protobuf)](https://developers.google.com/protocol-buffers/docs/proto3)).
```

Comment:
Is "design-first" the right description. The proto service description is more like a contract.

And when I hear design-first, I think of a VS designer GUI.

Reply:
I see both terms being used:
https://swagger.io/blog/api-design/design-first-or-code-first-api-development/
https://developers.redhat.com/blog/2018/07/12/contract-first-api-design-with-apicurio-and-red-hat-fuse/

@shirhatti do you have any preference?

Reply:
I do agree with James here that design first makes me think of a GUI with a designer. Let's use contract-first, it's more unambiguous.

Reply:
Updated.